src/setup/create/RestartRequired.tsx

```diff
@@ -1,13 +1,39 @@
 import { Button, Grid, Typography } from "@material-ui/core";
 import ArrowRightAltIcon from "@material-ui/icons/ArrowRightAlt";
 import CheckIcon from "@material-ui/icons/Check";
-import React, { ReactElement } from "react";
-import { restart$ } from "../../common/dockerUtil";
+import React, { ReactElement, useEffect } from "react";
+import { combineLatest, of } from "rxjs";
+import { mergeMap } from "rxjs/operators";
+import {
+  DockerSettings,
+  dockerSettings$,
+  isWSL2$,
+  restart$,
+} from "../../common/dockerUtil";
 import LinkToDiscord from "../LinkToDiscord";
 import RowsContainer from "../RowsContainer";
 import InfoBar from "./InfoBar";
 
 const RestartRequired = (): ReactElement => {
+  useEffect(() => {
+    combineLatest([dockerSettings$(), isWSL2$()])
+      .pipe(
+        mergeMap(([dockerSettings, isWSL2]) => {
+          const { wslEngineEnabled } = dockerSettings as DockerSettings;
+          console.log("wslEngineEnabled", wslEngineEnabled);
+          console.log("isWSL2", isWSL2);
+          if (wslEngineEnabled && !isWSL2) {
+            console.log(
+              "TODO: change wslEngineEnabled to false since only WSL 1 is supported"
+            );
+          }
+          return of(true);
+        })
+      )
+      .subscribe((r) => {
```

Comment:
Does the observable complete or does it have to be unsubscribed here?

Reply:
No need for cleanup here as the sources only emit 1 event and complete the stream.